plugin/storage/grpc/shared/interface.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-plugin"
+	"google.golang.org/grpc"
+
+	"github.com/jaegertracing/jaeger/proto-gen/storage_v1"
+	"github.com/jaegertracing/jaeger/storage/dependencystore"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+// StoragePluginIdentifier is the identifier that is shared by plugin and host.
+const StoragePluginIdentifier = "storage_plugin"
+
+// Handshake is a common handshake that is shared by plugin and host.
+var Handshake = plugin.HandshakeConfig{
+	MagicCookieKey:   "STORAGE_PLUGIN",
+	MagicCookieValue: "jaeger",
+}
+
+// PluginMap is the map of plugins we can dispense.
+var PluginMap = map[string]plugin.Plugin{
+	StoragePluginIdentifier: &StorageGRPCPlugin{},
+}
+
+// StoragePlugin is the interface we're exposing as a plugin.
+type StoragePlugin interface {
+	spanstore.Reader
+	spanstore.Writer
+	dependencystore.Reader
```

Comment:
What if we change this to 
```go
type StoragePlugin interface {
    SpanReader() spanstore.Reader
    SpanWriter() spanstore.Writer
    DependencyReader() dependencystore.Reader
}
```
?  This would make it easier to mock the plugin itself, since storage objects already have mocks.

Reply:
I'm not sure that it would make much difference although would make calls within GRPCServer itself a bit more verbose. For mocking we could just have an implementation that looks like
```
type MockStoragePlugin struct {
	spanstore.Reader
	spanstore.Writer
	dependencystore.Reader
}
```
and then inject the mocked storage objects directly, this would fulfil the interface and allow us to easily mock. I think? I'm unsure if I'm overlooking something here.